app/views/reports/regions/_header.html.erb

```diff
@@ -43,24 +48,45 @@
         aria-haspopup="true"
         aria-expanded="false"
       >
-        <%= @period.to_s %> report
+        <% if action_name == "cohort" %>
+          <%= cohort_report_type(@period) %>
+        <% else %>
+          <%= @period %>
+        <% end %>
       </a>
-      <div class="dropdown-menu" aria-labelledby="dropdownMenuLink">
-        <%= form_for(@period, url: request.params, method: :get, enforce_utf8: false) do |f| %>
-          <%= f.hidden_field :type, { id: "dropdown-period-type" } %>
-
-          <% (1..6).each do |num|
-              selected = false
-              date = Date.current.beginning_of_month.advance(months: -num)
-              period = Period.new(type: :month, value: date)
+      <% if action_name == "cohort" %>
+        <div class="dropdown-menu" aria-labelledby="dropdownMenuLink">
+          <% [:quarter, :month].each do |period_type|
+            period = Period.new(type: period_type, value: @period.to_date)
+            selected = true if period.type == @period.type && period == @period
```

Comment:
Hmmm, should we relax our `Period` comparator to return false if types are different?

Reply:
Yup did that.